File: programming/lab2/src/main/java/pokemons/Jirachi.java
package pokemons;

import moves.special.DazzlingGleam;
import moves.status.CalmMind;
import moves.status.Refresh;
import moves.status.ThunderWave;
import ru.ifmo.se.pokemon.Pokemon;
import ru.ifmo.se.pokemon.Type;

public class Jirachi extends ru.ifmo.se.pokemon.Pokemon {
    public Jirachi() {
        this("Jirachi", 1);
    }
    public Jirachi(String name, int level) {
        super(name, level);
        this.setLevel(level);
        this.setType(Type.STEEL, Type.PSYCHIC);
        this.setStats(100, 100, 100, 100, 100, 100);
        this.addMove(new DazzlingGleam());
        this.addMove(new Refresh());
        this.addMove(new CalmMind());
        this.addMove(new ThunderWave());
    }

}

File: programming/lab2/app/src/main/kotlin/org/itmo/pokemons/Jirachi.kt
package org.itmo.pokemons

import org.itmo.moves.special.DazzlingGleam
import org.itmo.moves.status.CalmMind
import org.itmo.moves.status.Refresh
import org.itmo.moves.status.ThunderWave
import ru.ifmo.se.pokemon.Pokemon
import ru.ifmo.se.pokemon.Type

class Jirachi(name: String, level: Int) : Pokemon(name, level) {
    init {
        setType(Type.STEEL, Type.PSYCHIC)
        setMove(CalmMind(), ThunderWave(), DazzlingGleam(), Refresh())
        setStats(100.0, 100.0, 100.0, 100.0, 100.0, 100.0)
    }
}